execution/gethexec/node.go

```diff
@@ -466,3 +471,86 @@ func (n *ExecutionNode) Synced() bool {
+	return n.ExecutionNode.DigestMessage(num, msg, msgForPrefetch)
+}
+func (n *ExecutionClientImpl) Reorg(count arbutil.MessageIndex, newMessages []arbostypes.MessageWithMetadataAndBlockHash, oldMessages []*arbostypes.MessageWithMetadata) containers.PromiseInterface[[]*execution.MessageResult] {
+	return n.ExecutionNode.Reorg(count, newMessages, oldMessages)
+}
+func (n *ExecutionClientImpl) HeadMessageNumber() containers.PromiseInterface[arbutil.MessageIndex] {
+	return n.ExecutionNode.HeadMessageNumber()
+}
+func (n *ExecutionClientImpl) HeadMessageNumberSync(t *testing.T) containers.PromiseInterface[arbutil.MessageIndex] {
+	return n.ExecutionNode.HeadMessageNumberSync(t)
+}
+func (n *ExecutionClientImpl) ResultAtPos(pos arbutil.MessageIndex) containers.PromiseInterface[*execution.MessageResult] {
+	return n.ExecutionNode.ResultAtPos(pos)
+}
+func (n *ExecutionClientImpl) MarkFeedStart(to arbutil.MessageIndex) containers.PromiseInterface[struct{}] {
+	return n.ExecutionNode.MarkFeedStart(to)
+}
+
+// ExecutionRecorder interface implementation
+func (n *ExecutionClientImpl) RecordBlockCreation(
+	ctx context.Context,
+	pos arbutil.MessageIndex,
+	msg *arbostypes.MessageWithMetadata,
+) (*execution.RecordResult, error) {
+	panic("RecordBlockCreation not supported")
+}
+func (n *ExecutionClientImpl) MarkValid(pos arbutil.MessageIndex, resultHash common.Hash) {
+	panic("MarkValid not supported")
+}
+func (n *ExecutionClientImpl) PrepareForRecord(ctx context.Context, start, end arbutil.MessageIndex) error {
+	panic("PrepareForRecord not supported")
+}
+
+// ExecutionSequence interface implementation
+func (n *ExecutionClientImpl) Pause() {
+	panic("Pause not supported")
+}
+func (n *ExecutionClientImpl) Activate() {
+	// panic("Activate not supported")
+}
+func (n *ExecutionClientImpl) ForwardTo(url string) error {
+	panic("ForwardTo not supported")
+}
+func (n *ExecutionClientImpl) SequenceDelayedMessage(message *arbostypes.L1IncomingMessage, delayedSeqNum uint64) error {
+	panic("SequenceDelayedMessage not supported")
+}
+func (n *ExecutionClientImpl) NextDelayedMessageNumber() (uint64, error) {
+	panic("NextDelayedMessageNumber not supported")
+}
+func (n *ExecutionClientImpl) Synced() bool {
+	panic("Synced not supported")
+}
+func (n *ExecutionClientImpl) FullSyncProgressMap() map[string]interface{} {
+	panic("FullSyncProgressMap not supported")
+}
+
+// Other funcs from FullExecutionClient
+// not thread safe
+func (n *ExecutionClientImpl) Start(ctx context.Context) error {
+	return n.ExecutionNode.Start(ctx)
+}
+func (n *ExecutionClientImpl) StopAndWait() {
+	n.ExecutionNode.StopAndWait()
+}
+func (n *ExecutionClientImpl) Maintenance() error {
+	return n.ExecutionNode.Maintenance()
+}
+func (n *ExecutionClientImpl) ArbOSVersionForMessageNumber(messageNum arbutil.MessageIndex) (uint64, error) {
```

Comment:
I think this one is only used by batchposter and shouldn't be needed here?
Otherwise - it should be moved to ExecutionClient interface